rocky/rocky/settings.py

```diff
@@ -274,8 +293,33 @@
 # https://docs.djangoproject.com/en/4.2/howto/static-files/
 
 STATIC_URL = "/static/"
-STATIC_ROOT = BASE_DIR / "static"
+STATIC_ROOT = env.path("STATIC_ROOT", BASE_DIR / "static")
 STATICFILES_DIRS = (BASE_DIR / "assets",)
+STATICFILES_FINDERS = [
+    "django.contrib.staticfiles.finders.FileSystemFinder",
+    "django.contrib.staticfiles.finders.AppDirectoriesFinder",
+    "compressor.finders.CompressorFinder",
+]
+COMPRESS_OFFLINE = True
+COMPRESS_STORAGE = "compressor.storage.BrotliCompressorFileStorage"
+
+STORAGES = {
+    "staticfiles": {
+        "BACKEND": "whitenoise.storage.CompressedManifestStaticFilesStorage",
+    },
+}
+
+
+def immutable_file_test(path, url):
+    # Match filename with 12 hex digits before the extension e.g.
+    # app.db8f2edc0c8a.js. Confifguraring this is necessary because whitenoise
+    # doesn't automatically detect the django-compressor files as immutable.
+    return re.match(r"^.+\.[0-9a-f]{12}\..+$", url)
```

Comment:
You might want to compile this pattern (outside of the function, e.g. as `_IMMUTABLE_FILE_TEST_PATTERN = re.compile(...)`)